### PR TITLE
Add workflows to enforce codestyle

### PR DESCRIPTION
**Pull request by bruno-delfino1995**

### CODE REVIEW

File: .github/workflows/clippy-checks.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
```
Comment (thepabloaguilar) [inline .github/workflows/clippy-checks.yml:21]: I think we don't need to keep the license here!

Comment (bruno-delfino1995) [reply .github/workflows/clippy-checks.yml:21]: I didn't write this code but copied and pasted it. The source of it is [here](https://github.com/CertainLach/jrsonnet).

One of the conditions for copies of code licensed under  [MIT](https://choosealicense.com/licenses/mit/) is to keep the copyright notice. If we were to remove it, we would be breaking the license and not giving proper credits to the author.

Comment (thepabloaguilar) [reply .github/workflows/clippy-checks.yml:21]: Seems that you've made your homework!!
Great!! 🎉 🎉 🎉 🎉

**Review by thepabloaguilar — APPROVED** [review]
LGTM

**Pull request merged**